src/utils/price_estimator.ts

```diff
@@ -0,0 +1,4 @@
+export interface PriceEstimator {
+	getARPriceForByteCount: (byteCount: number) => Promise<number>;
```

Comment:
Was there a good reason to not return a Promise<ARPriceData> which has Winston value validations built in?

Reply:
The value this function returns is an AR value, which currently doesn't exist on ARDataPrice. I was following the pattern of the original get function in the class, which returned a number as well.

Reply:
It would have been pretty easy and tidy to just add an ar computed property to the ARPriceData class which could use an arPerWinston constant like so:
```
const arPerWinston = 0.000_000_000_001;

export class ARDataPrice {
	/**
	 * @returns an ARDataPrice instance with the given byte count and Winston amount
	 * @throws {@link Error} if negative or non-integer values are provided for either value
	 */
	constructor(readonly numBytes: number, readonly winstonPrice: number) {
		if (numBytes < 0 || !Number.isInteger(numBytes) || winstonPrice < 0 || !Number.isInteger(winstonPrice)) {
			throw new Error(
				`numBytes (${numBytes}) and winstonPrice (${winstonPrice}) should be non-negative integer values.`
			);
		}

		this.numBytes = numBytes;
		this.winstonPrice = winstonPrice;
	}

	get ar(): number {
		return arPerWinston * this.winstonPrice;
	}
}
```